mula/scheduler/repositories/sqlalchemy/datastore.py

```diff
@@ -55,3 +57,22 @@ def __init__(self, dsn: str) -> None:
         self.session = orm.sessionmaker(
             bind=self.engine,
         )
+
+
+def retry(max_retries=3, retry_delay=5):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            for i in range(max_retries):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    if i == max_retries - 1:
+                        raise e
+
+                    print(f"Retrying {func.__name__} in {retry_delay} seconds ({i+1}/{max_retries}): {str(e)}")
```

Comment:
Perhaps a log statement instead of print would be better?

Reply:
`str(e)` isn't necessary, the f-string will always use the stringified version of objects